kiwi/system/profile.py

```diff
@@ -53,6 +53,18 @@ def __init__(self, xml_state):
         self._drivers_to_profile()
         self._root_partition_uuid_to_profile()
 
+    def get(self):
+        """
+        Return all profile elements that has a value
+        """
+        profile = {}
+        for key, value in list(self.dot_profile.items()):
+            if value:
+                profile[format(key)] = self._format(value)
```

Comment:
This `format` doesn't do anything useful. And also this is a bug: it won't export value with an empty string, which is incorrect: empty string **is** a value. The `_format` function doesn't care about `False` to "false" either.

I would also start adding annotations.

Reply:
> This format doesn't do anything useful

yes, absolutely

> it won't export value with an empty string, 

yes, I think this was done intentional because the initrd code that makes use of the variables does not handle variables with no value. But I will allow it in the _format helper now as the origin of the content for those variables comes from the image description and all constraints for them are applied earlier.

> I would also uppercase it to be consistent with the environment overall

I can't do this because it would break the code that uses the variables. This would be the case in the dracut code for kiwi and it would break for users that use our variables in their custom scripts. Thus I'm afraid can't uppercase them. You should have in mind the variables you see here are from a .profile file which we offer to people to be sourced in their custom scripts. Also the same .profile file will be placed in the dracut modules and sourced there as well. Thus I have to care for not breaking this.

Reply:
Uppercasing it is just "to be nice", but if that might break stuff, yeah, better not. Still, out of curiosity, why it cannot be done? You suppose to call any arbitrary script (`disk.sh` in this case, but so valid to all others) in their _own space_ (look at the bug below I pointed you out). So that said, you are exporting all the environment _exclusively to that script_, which should by no means never affect anything outside of it.

At least that would be my design...

Reply:
> f you done it right, then the script disc.sh should see KIWI_STUFF and dracut should keep seeing kiwi_stuff. To do so, I would:

yes this is correct and we can uppercase the variables and keep the .profile file contents as they are right now. But wouldn't that be inconsistent ? Given that I can't change to uppercase in the .profile file for compat reasons we would need to stick with that difference for a long time, or am I mistaken